### PR TITLE
gevent: remove implementation dependent tests

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
These tests depended on the implementation details of the context
management for gevent. There exist very similar tests that assert the
correct context management behaviour which are less brittle.

(Part of #1936)

### CODE REVIEW

File: tests/contrib/gevent/test_tracer.py [hunk 1]
```diff
@@ -40,75 +40,6 @@ def tearDown(self):
...
-
-    def test_main_greenlet_context(self):
-        # the main greenlet must have a ``Context`` if called
-        ctx_tracer = self.tracer.get_call_context()
-        main_greenlet = gevent.getcurrent()
-        ctx_greenlet = getattr(main_greenlet, "__datadog_context", None)
-        assert ctx_tracer is ctx_greenlet
-        assert len(ctx_tracer._trace) == 0
-
-    def test_get_call_context(self):
```
Comment (Kyle-Verhoog) [inline tests/contrib/gevent/test_tracer.py:57]: `test_trace_greenlet` is a very similar test which provides a stronger assertion

File: tests/contrib/gevent/test_tracer.py [hunk 2]
```diff
@@ -40,75 +40,6 @@ def tearDown(self):
...
-        # a greenlet will not have a context if the tracer is not used
-        def greenlet():
-            gevent.sleep(0.01)
-
-        g = gevent.spawn(greenlet)
-        g.join()
-        ctx = getattr(g, "__datadog_context", None)
-        assert ctx is None
-
-    def test_spawn_greenlet(self):
```
Comment (Kyle-Verhoog) [inline tests/contrib/gevent/test_tracer.py:89]: This is literally a duplicate of `test_get_call_context` above.

File: tests/contrib/gevent/test_tracer.py [hunk 3]
```diff
@@ -40,75 +40,6 @@ def tearDown(self):
...
-        def greenlet():
-            self.tracer.get_call_context()
-
-        g = gevent.spawn(greenlet)
-        g.join()
-        ctx = getattr(g, "__datadog_context", None)
-        assert ctx is not None
-        assert 0 == len(ctx._trace)
-
-    def test_spawn_later_greenlet(self):
```
Comment (Kyle-Verhoog) [inline tests/contrib/gevent/test_tracer.py:100]: `test_trace_later_greenlet` is similar but with a stronger assertion.

File: tests/contrib/gevent/test_tracer.py [hunk 4]
```diff
@@ -40,75 +40,6 @@ def tearDown(self):
...
-            return self.tracer.get_call_context()
-
-        g = gevent.spawn(greenlet)
-        g.join()
-        ctx = g.value
-        stored_ctx = getattr(g, "__datadog_context", None)
-        assert stored_ctx is not None
-        assert ctx == stored_ctx
-
-    def test_get_call_context_twice(self):
```
Comment (Kyle-Verhoog) [inline tests/contrib/gevent/test_tracer.py:69]: Added a test case to cover this functionality

**Review by jd — APPROVED** [review]

**Pull request merged**